src/logic/calculate.js

```diff
@@ -0,0 +1,71 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  let { total, currentValue, operation } = dataObject;
+
```

Comment:
dataObject should contain `total`, `next`, `operation`

Reply:
i fix this iisue